### ISSUE TITLE
Add get_angle_x etc. methods to Quaternion

### ISSUE THREAD

**Issue by alice-i-cecile** [bitshifter/glam-rs#226]
This is the natural complement to [`from_rotation_x`](https://docs.rs/bevy/0.5.0/bevy/math/struct.Quat.html#method.from_rotation_x) and will help users comfortably extract the information they need.

**Comment by bitshifter** [comment]
I don't think it is a complement of `from_rotation_x` in that `from_rotation_x` creates a quaternion which rotates solely around the X axis. However `get_angle_x` will be operating on an arbitrary rotation, so would it extract the amount the quaternion rotates around +X from an arbitrary rotation axis? Could you provide an example use case of why someone might want this?

**Comment by alice-i-cecile** [comment]
Sure! So, the exact mechanism I had in mind was:

1. Project the quaternion vector to the `xy` plane.
2. Find the angle that this makes with the cardinal `x` direction.

Now that I write this out, it's apparent that the method should actually be `get_angle_xy` and so on.

I ran into this when trying to help another user figure out the current angle that their space-ship was pointing in a 2D asteroids game. From poking at the docs again with a fresh set of eyes, it seems that https://docs.rs/glam/0.17.3/glam/f32/struct.Quat.html#method.to_axis_angle may give us the information we want in some of these cases?

The full, 3D-aware version described would be useful for something like "I have a cannon in 3D, and want to show the direction it's pointing to in a top down way". Now that I've come back to it, I think that I would actually like:

1. Stronger docs on `to_axis_angle`, or perhaps `Quaternion` itself, that are more beginner friendly and explain the basic terminology.
2. A `project_to_plane` method, which can then be composed with `angle_between` when necessary.

**Comment by bitshifter** [comment]
Was this discussion on discord? Can you link it?

This use case sounds like something I'd typically use vector dot products for and acos of the angle was required.

**Comment by alice-i-cecile** [comment]
It moved to DMs unfortunately. I've passed a link on to this thread; hopefully they're comfortable sharing a bit more about their experience and use cases here :)

> This use case sounds like something I'd typically use vector dot products for and acos if the angle was required.

Generally I would take the same approach. That level of linear algebra mastery is often a bit much for the end user who's new to making games though; particularly if they're working in 2D. My hope is that we can work to expose common, simple operations with nicely named methods in glam and bevy, and wrap this linear algebra under the hood.

Of course, that may ultimately be an argument in favor of a 2D API for Bevy: using quaternions at all very much feels like needless pain.

**Comment by KoomZog** [comment]
This discussion came up when I asked on the Bevy Discord "Help" channel about an issue I was having.
 
To preface this question and how I decided to tackle it: I'm not a very experienced programmer. I just found out about Bevy 7 days ago, and decided to give it a go. I've never worked with Rust before and I have never used Quaternions before. I'm definitely in the category alice-i-cecile mentions:

>That level of linear algebra mastery is often a bit much for the end user who's new to making games

The game I'm making to learn Bevy is a clone of the old windows game Comet Busters. It's a top down 2D space themed game where the player controls a spaceship and the objective is to shoot at asteroids. The ship can rotate (z axis) and accelerate. It also has a shield that can be activated if the ship is on collision course with an asteroid.

I didn't really understand how Quaternions work (still don't, but at least a little bit now). This gave me some headache when I tried to make the ship rotate with the `transform.rotate()` function. I think this was about when I asked for help in the Discord channel. Alice answered and I sent her my code in a DM and we tried to figure out what I had messed  up. We made some progress, but we didn't solve the whole thing. I wanted to get the z rotation of the ship entity in radians, so that I could use sin/cos to accelerate in the direction it was pointing. This was the origin of this issue.

Since Quaternions gave me nothing but headache, I decided the next day to take a pragmatic approach and not bother with learning Quaternions for now. Instead, I created my own `z_rotation` component which turns the ship with `transform.rotation = Quat::from_rotation_z(z_rotation.0);`. This was quite simple to work with and works well for my application. I would definitely welcome a simpler approach to 2D rotation in the API and after my encounter with quaternions I agree with alice-i-cecile:

>Of course, that may ultimately be an argument in favor of a 2D API for Bevy: using quaternions at all very much feels like needless pain.

I don't know what would be a good intuitive approach for that, and I'm not experienced enough to propose any kind of solution. I'm just here to share my experience.

**Comment by bitshifter** [comment]
@alice-i-cecile I've made an example which performs 2d rotations in a few different ways here https://github.com/bitshifter/bevy-2d-rotation-example/blob/main/src/main.rs. 

There are 3 different kinds of rotation in there:

* rotating and moving a player ship at a fixed linear and angular velocity
* rotating an enemy ship to face the player ship at a fixed angular velocity
* rotating an enemy ship to face the player ship immediately

I feel like that should cover most basic use cases.

I did extend `Quat` functionality in that example with a `from_rotation_arc_2d` method although it's more of an optimization, the current `Quat::from_rotation_arc` works fine as well, Other than that, nothing leaps out to me as being useful to move into `glam` as helper code. 

Perhaps this would be a good thing to add to bevy's examples though?
